app/components/main-content/PrelaunchSignUpForm.tsx

```diff
@@ -92,10 +94,11 @@ const PrelaunchSignUpForm: React.FC<PrelaunchSignUpFormProps> = ({
             <UniversalButton
               id="join-waiting-list-button"
               type="submit"
-              label="Join Waiting List"
+              label={isLoading ? "Adding..." : "Join Waiting List"}
```

Comment:
I like this implementation for visually indicating the loading state to users. 

But I'm curious how this will be experienced when using a screen reader. We could potentially render the label text within an `aria-live` element. We could also set `aria-label` conditionally - Mantine doesn't use this, so we wouldn't be overriding anything.

Reply:
I just tested this using Mac's VoiceOver screen reader. It does announce the change to "Adding..." when the button is clicked! I imagine because the button still has focus after being clicked. 

`aria-label` could be used to be a bit more descriptive than "Adding...", like "Adding you to our waiting list", if desired.

Reply:
Thanks for testing it! The button is actually quite small so including the whole sentence will make the fonts very small. Maybe I will let @ChTiSh to decide when she's styling the home page to see which option is better. 

For now should I add an aria-label or the screen reader can read the label on the button?

Reply:
Aria-label added to button.